clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1946,6 +1969,20 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     return true;
   }
 
+  // Generate kernel argument to intialize specialization constants. This
+  // argument is only generated when the target has no native support for
+  // specialization constants
+  void handleSyclKernelHandlerType() {
+
+    ASTContext &Context = SemaRef.getASTContext();
+    llvm::Triple T = Context.getTargetInfo().getTriple();
+    if (T.isSPIR() && T.getSubArch() == llvm::Triple::NoSubArch)
```

Comment:
I'd love it if @mdtoguchi  could double-check this for us.

Reply:
If I understand it correctly, this check filters spir64-unknown. I confirmed this with @mdtoguchi offline. It is to differentiate behavior for targets with native support for specialization constants vs no native support. My understanding is that only spir64-unknown has native support. @bader can you please confirm?

Reply:
Sounds right to me. `isSPIR()` returns `true` for `spir`/`spir64` targets and it's subtargets like `spir64_gen` (which we use in AOT mode). `T.getSubArch() == llvm::Triple::NoSubArch` condition excludes AOT subtargets and allows only `spir`/`spir64` targets.

Reply:
ok. Thank you for clarification!